testservices/novaservice/service.go

```diff
@@ -304,6 +305,16 @@ func (n *Nova) server(serverId string) (*nova.ServerDetail, error) {
 	if !ok {
 		return nil, testservices.NewServerByIDNotFoundError(serverId)
 	}
+	groups := n.allServerSecurityGroups(serverId)
+	if len(groups) > 0 {
+		groupNames := make([]nova.SecurityGroupName, 0)
```

Comment:
You can leave out the 0 here. Alternatively if you want to avoid growing the slice you can specify the length of groups:
```go
groupNames := make([]nova.SecurityGroupName, len(groups))
for i, group := range groups {
    groupNames[i] = nova.SecurityGroupName{Name: group.Name})
}
```